src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -263,6 +264,22 @@ public static Component getPlayerTabListDisplayName(Player player)
         return event.getDisplayName();
     }
 
+    @Nullable
+    public static MutableComponent getPlayerJoinMessage(Player player)
+    {
+        PlayerEvent.JoinMessageFormat event = new PlayerEvent.JoinMessageFormat(player);
+        MinecraftForge.EVENT_BUS.post(event);
+        return event.isCanceled() ? null : event.getJoinMessage();
+    }
+
+    @Nullable
+    public static MutableComponent getPlayerLeaveMessage(Player player)
+    {
+        PlayerEvent.LeaveMessageFormat event = new PlayerEvent.LeaveMessageFormat(player);
+        MinecraftForge.EVENT_BUS.post(event);
+        return  event.isCanceled() ? null : event.getLeaveMessage();
+    }
+
```

Comment:
What are the point of these if they aren't used? The best way to do this is to get the original mutable component and pass it in as part of the event and then check the event before sending.